### PR TITLE
Add Nix flake support

### PR DESCRIPTION
**Pull request by emctoo**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -37,14 +38,14 @@ mypy = ">=0.950"
...
-ruff = "^0.4.3"
+ruff = "^0.4.6"
 
 [build-system]
 requires = ["poetry-core>=1.0.0"]
 build-backend = "poetry.core.masonry.api"
 
 [tool.mypy]
-python_version = 3.7
+# python_version = 3.7
```
Comment (xoolive) [inline pyproject.toml:48]: why? ok to update version to 3.9 though

File: src/tangram/app.py [hunk 2]
```diff
@@ -43,7 +44,8 @@ async def shutdown_publish_job() -> None:
     log.info("shutdown - publish job done")
 
 
-templates = Jinja2Templates(directory="templates")
+tangram_module_root = pathlib.Path(__file__).resolve().parent
+templates = Jinja2Templates(directory=tangram_module_root / "templates")
```
Comment (xoolive) [inline src/tangram/app.py:48]: can we do something similar for the log.yml file?

**Comment by xoolive** [comment]
merging it, but please keep an eye on my comment!
many thanks!

File: src/tangram/app.py (hunk 2 above)
Comment (emctoo) [reply src/tangram/app.py:48]: @xoolive Sorry, missed this.

I'm thinking of adding a config file, `.toml` for example, so plugins can have their sections too. And by default it's placed in the package root directory.

**Pull request merged**